docs/source/distributed.rst

```diff
@@ -516,6 +492,67 @@ Collective functions
 
     :class:`~torch.distributed.ReduceOp` is recommended to use instead.
 
+
+Distributed Key-Value Store
+---------------------------
+
+The distributed package comes with a distributed key-value store, which can be
+used to share information between processes in the group as well as to
+initialize the distributed package in
+:func:`torch.distributed.init_process_group` (by explicitly creating the store
+as an alternative to specifying ``init_method``.) There are 3 choices for
+Key-Value Stores: :class:`~torch.distributed.TCPStore`,
+:class:`~torch.distributed.FileStore`, and :class:`~torch.distributed.HashStore`.
+
+.. autoclass:: Store
+    :members:
+    :special-members:
+
+.. autoclass:: TCPStore
+    :members:
+    :special-members: __init__
+
+.. autoclass:: HashStore
+    :members:
+    :special-members: __init__
+
+.. autoclass:: FileStore
+    :members:
+    :special-members: __init__
+
+.. autoclass:: PrefixStore
+    :members:
+    :special-members: __init__
+
+
+
+Object Oriented Process Groups
+------------------------------
+
+We recommend most models use the global ProcessGroup registration and dist.*
+APIs but if you need more control over the process groups (i.e. dynamic world
+sizes) you can directly instantiate them.
+
+There are some notable differences:
+
+* These APIs are always asynchronous and require you to manually synchronize the returned :class:`~torch.distributed.Work` objects.
```

Comment:
Can't users still manually create a PG and then call `dist.all_reduce(group=pg)` and get the same thing, but without skipping our validation layers that are only present in the upper layer?

I'm still wondering in what cases it would be good to call `pg.all_reduce` instead.  (Granted, my other PRs for adding 'group_src'/'group_dst' are needed for this statement to hold.)

Also, I haven't looked at it carefully but I am worried about manual PG creation skipping our UUID logic.  I think that ends up not mattering if you use a prefix store for each pg, but if you use the same prefix store for 2 PGs by accident you'd be in trouble right?  Should we actually encourage users to use ProcessGroup ctor?  or should we improve our `new_group` type APIs to let the PGs be created without a harmful tie to the world?

Reply:
agreed, removed this documentation in favor of your other PRs